DevIncubator-Autopark/Vehicle.cs

```diff
@@ -29,11 +31,25 @@ internal class Vehicle : IComparable<Vehicle>
             TankCapacity = tankCapacity;
         }
 
-        public double GetCalcTaxPerMonth() => (Weight * WeightCoefficient) + (VehicleType.TaxCoefficient * TaxCoefficient) + TaxChange;
-
-        public override string ToString() => $"{VehicleType},{Model},{LicensePlate},{Weight}," +
+        public override string ToString() => $"{VehicleType}, {VehicleEngine.TypeName}, {Model},{LicensePlate},{Weight}," +
             $"{YearIssue},{Mileage},{Color},{TankCapacity},{GetCalcTaxPerMonth().ToString("0.00")}";
 
+        public override bool Equals(object obj)
+        {
+            if ((obj == null) || !GetType().Equals(obj.GetType()))
```

Comment:
You can write this method just in one line. Use `obj is Vehicle second &&..` for example.

Reply:
It's `pattern matching` approach
obj is Type variable

Reply:
fixed